### PR TITLE
Initial module work

### PR DESCRIPTION
**Pull request by JayH5**
There are a few Puppet modules out there for managing Marathon but I think we can do better.

The existing modules are mostly:
- Not well-maintained/haven't been updated in a while
- Not super-well written or written to achieve different goals
- Don't manage logging
- Don't manage JAVA_OPTS for things like setting heap size

This is a fairly basic module that reuses a few things from the deric/mesos module. There are no actual tests here yet but Puppet lint runs at least.

### CODE REVIEW

File: Puppetfile [hunk 1]
```diff
@@ -0,0 +1,3 @@
+mod 'puppetlabs/stdlib', '4.10.0'
+mod 'puppetlabs/apt', '2.2.1'
+mod 'deric/mesos', '0.6.4'
```
Comment (jerith) [inline Puppetfile:3]: Do we need these exact versions?

Comment (JayH5) [inline Puppetfile:3]: For `r10k`, yes.

However, I was thinking of switching to [`librarian-puppet`](http://librarian-puppet.com) which has a similar Puppetfile but is much more like Bundler/pip in that you can specify a range of valid versions and it actually does dependency resolution.

Wasn't sure whether to do that here or in a new PR.

Comment (jerith) [inline Puppetfile:3]: Let's do it in a new PR.

**Comment by jerith** [comment]
:+1:
Let's merge this and see how well it works. We can fix it if it's broken.

**Pull request merged**